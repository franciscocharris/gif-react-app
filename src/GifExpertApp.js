import React, {useState} from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

// esto es mas por la prueba
// export const GifExpertApp = ( {defaultCategories = []} ) => {

//     // const categories = ['one punch', 'naruto', 'dragon ball'];
//     // const [categories, setCategories] = useState(['naruto']);
//     const [categories, setCategories] = useState(defaultCategories);
export const GifExpertApp = ( ) => {

    // const categories = ['one punch', 'naruto', 'dragon ball'];
    const [categories, setCategories] = useState(['naruto']);

    // const handleApp = () => {
    //     // se usa el spread operator paras hacer un duplicado del valor y poder a√±adirle uno nuevo, basicamente
    //     // se creo otro arreglo
    //     setCategories( [...categories, 'hola mundo react'] );
    // }
    
    return (
        <>
            <div className='head' >
                <h1>GifExpert App</h1> 
                <a href='https://www.facebook.com/profile.php?id=100013541311546' className='credito'>por: Francisco M. Charris C.</a>
            </div>
            <AddCategory setCategories={setCategories} />
            <hr />

            <ol>
               {
                    categories.map( category => 
                        <GifGrid 
                        // enviar props
                            key={category}
                            category={category}
                        />    
                    )
               }
            </ol>
        </>
    );
}