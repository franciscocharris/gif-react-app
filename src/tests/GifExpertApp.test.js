import { shallow } from 'enzyme';
import React from 'react';
import { GifExpertApp } from '../GifExpertApp';

describe('pruebas en el componentye padre GifExpertApp', () => {
    const wrapper = shallow(<GifExpertApp />);

    test('debe coincidir con la fotografia', () => {
        expect(wrapper).toMatchSnapshot();
    });

    // test('debe tener la misma cantidad de GifGrid que uno manda', () => {
    //     const defaultCategories = ['naruto', 'dragon Ball'];
    //     const wrapper = shallow(<GifExpertApp defaultCategories={defaultCategories} />);

    //     expect(wrapper.find('GifGrid').length).toBe( defaultCategories.length);
    // })
    
    
})
