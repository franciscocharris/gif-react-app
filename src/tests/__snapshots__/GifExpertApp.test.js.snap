// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en el componentye padre GifExpertApp debe coincidir con la fotografia 1`] = `
<Fragment>
  <div
    className="head"
  >
    <h1>
      GifExpert App
    </h1>
    <a
      className="credito"
      href="https://www.facebook.com/profile.php?id=100013541311546"
    >
      por: Francisco M. Charris C.
    </a>
  </div>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
